src/main/PasswordVerifier.java

```diff
@@ -0,0 +1,110 @@
+package main;
+
+import java.util.regex.Pattern;
+
+public class PasswordVerifier {
+
+    public PasswordVerifier() {
+
+    }
+
+    public boolean verify(String password) {
+        int numValid = 0;
+
+        // password is never OK if item 1.4 (password should have one lowercase letter at least) is not true.
+        try {
+            if (containsLowercase(password)) numValid++;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e);
+            return false;
+        }
+
+        try {
+            if (isNotNull(password)) numValid++;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e);
+        }
+
+        try {
+            if (isLargerThanEight(password)) numValid++;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e);
+        }
+
+        if (numValid >= 3) return true; // for optimization
+
+        try {
+            if (containsUppercase(password)) numValid ++;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e);
+        }
+
+        if (numValid >= 3) return true; // for optimization
+
+        try {
+            if (containsDigit(password)) numValid ++;
+        } catch (IllegalArgumentException e) {
+            System.out.println(e);
+        }
```

Comment:
All these try catch statements do the same thing. Can we simplify to something like:

```java
try {
    if (!containsLowercase(password)) return false
    numValid++
    if (isNotNull(password)) numValid++;
    if (isLargerThanEight(password)) numValid++;
    if (numValid >= 3) return true; // for optimization
    if (containsUppercase(password)) numValid ++;
    if (numValid >= 3) return true; // for optimization
    if (containsDigit(password)) numValid ++;
} catch (IllegalArgumentException e) {
    System.out.println(e);
}

```